d2ir/compile.go

```diff
@@ -128,6 +86,22 @@ func (c *compiler) compileField(dst *Map, kp *d2ast.KeyPath, refctx *RefContext)
 				parent: f,
 			}
 		}
+		switch NodeBoardKind(f) {
+		case BoardScenario:
+			c.overlay(dst.Parent().Parent().(*Map), f)
```

Comment:
Why is this `dst.Parent().Parent()`? `f` here would be the root of the board and so shouldn't it be `f.Parent().Parent()`?

Reply:
```
...
scenarios: {
  1: {
     ...
  }
}
```

`f` is the field `1`, `scenarios` map is the `dst`. 

so `dst.Parent()` gets the field `scenarios`
and then `dst.Parent().Parent()` gets the map that the field `scenarios` is contained in, which is what we want.

i can change to use `f`, but tests only pass if it's `f.Parent().Parent().Parent()`

Reply:
Right. The problem with using `dst` is that you don't know what it actually is. You could have `scenarios.x: {}` where now `dst` is the base map.

I think the best solution is `ParentField(ParentField(f)).Map()` to grab the base map.

Reply:
`d2ir.ParentField()` I mean

Reply:
oh i didn't know there were that many helpers. updated to use ParentBoard and stuff